src/components/UpgradePlanForm/UpgradePlanForm.js

```diff
@@ -30,98 +31,104 @@ class UpgradePlanForm extends React.Component {
     this.setState({ availablePlans: ClientTypePlans });
   }
 
-  async handleSubmit(event) {
-    event.preventDefault();
+  async onSubmit(values, { setSubmitting }) {
     await this.dopplerLegacyClient.sendEmailUpgradePlan({
-      IdClientTypePlanSelected: this.state.formData[fieldNames.selectedPlanId],
-      Detail: this.state.formData[fieldNames.message],
+      IdClientTypePlanSelected: values[fieldNames.selectedPlanId],
+      Detail: values[fieldNames.message],
     });
+    setSubmitting(false);
     this.props.handleClose();
   }
 
-  changeHandler = (event) => {
-    const name = event.target.name;
-    const value = event.target.value;
-
-    this.setState((prevState) => ({
-      formData: {
-        ...prevState.formData,
-        [name]: value,
-      },
-    }));
-
-    this.setState((prevState) => ({
-      formIsValid: !!prevState.formData[fieldNames.message],
-    }));
-  };
+  validate(values) {
+    const errors = {};
+    if (!values[fieldNames.message]) {
+      // TODO: translate it, here or in the markup
+      errors[fieldNames.message] = 'Required';
+    }
+    return errors;
+  }
 
   render() {
     const {
-      changeHandler,
-      handleSubmit,
-      state: { availablePlans, formIsValid, formData },
+      onSubmit,
+      validate,
+      state: { availablePlans },
       props: { handleClose },
     } = this;
     const isUserDataLoaded = !!availablePlans;
+    // TODO: what happens when availablePlans is an empty array?
+    const firstPlanId = availablePlans && availablePlans.length && availablePlans[0].IdUserTypePlan;
```

Comment:
It is to fix this bug:

![image](https://user-images.githubusercontent.com/1157864/54571232-88d94680-49c0-11e9-8513-9d5c6ac49b4c.png)